reconcile/utils/oc_map.py

```diff
@@ -0,0 +1,253 @@
+    """
+
+    def __init__(
+        self,
+        connection_parameters: Iterable[OCConnectionParameters],
+        clusters_untyped: Optional[Iterable[MutableMapping[Any, Any]]] = None,
+        namespaces_untyped: Optional[Iterable[MutableMapping[Any, Any]]] = None,
+        integration: str = "",
+        e2e_test: str = "",
+        settings_untyped: Optional[Mapping[Any, Any]] = None,
+        internal: Optional[bool] = None,
+        use_jump_host: bool = True,
+        thread_pool_size: int = 1,
+        init_projects: bool = False,
+        init_api_resources: bool = False,
+        cluster_admin: bool = False,
+    ):
+        self._oc_map: dict[str, Union[OCDeprecated, OCLogMsg]] = {}
+        self._privileged_oc_map: dict[str, Union[OCDeprecated, OCLogMsg]] = {}
+        self._calling_integration = integration
+        self._calling_e2e_test = e2e_test
+        self._internal = internal
+        self._use_jump_host = use_jump_host
+        self._thread_pool_size = thread_pool_size
+        self._init_projects = init_projects
+        self._init_api_resources = init_api_resources
+        self._lock = Lock()
+        self._jh_ports: dict[str, int] = {}
+        self._settings_dict = settings_untyped
+
+        # TODO: remove these once jumphosts are typed
+        # ############################################
+        self._jumphosts_dict: dict[Any, Any] = {}
+        for cluster_dict in clusters_untyped or []:
+            self._jumphosts_dict[cluster_dict.get("name")] = cluster_dict.get(
+                "jumpHost"
+            )
+        for ns_dict in namespaces_untyped or []:
+            cluster_d = ns_dict.get("cluster")
+            if cluster_d:
+                self._jumphosts_dict[cluster_d.get("name")] = cluster_d.get("jumpHost")
+        # ############################################
+
+        # init a namespace with clusterAdmin with both auth tokens
+        # OC_Map is used in various places and even when a namespace
+        # declares clusterAdmin token usage, many of those places are
+        # happy with regular dedicated-admin and will request a cluster
+        # with oc_map.get(cluster) without specifying privileged access
+        # specifically
+        privileged_clusters: list[OCConnectionParameters] = [
+            c for c in connection_parameters if (c.is_cluster_admin or cluster_admin)
+        ]
+        unprivileged_clusters: list[OCConnectionParameters] = [
+            c
+            for c in connection_parameters
+            if not (c.is_cluster_admin or cluster_admin)
+        ]
+
+        threaded.run(
+            self.init_oc_client,
+            unprivileged_clusters,
+            self._thread_pool_size,
+            privileged=False,
+        )
+        threaded.run(
+            self.init_oc_client,
+            privileged_clusters,
+            self._thread_pool_size,
+            privileged=True,
+        )
+
+    def set_jh_ports(self, jh: MutableMapping[Any, Any]) -> None:
```

Comment:
shouldn't this be private?

Reply:
Fixed :+1: 

Also fixed in `disabled()` function now